src/runtime/getPositions.ts

```diff
@@ -54,23 +64,29 @@ export async function getBaseTokensInfo(
   getTokensInfoUrl: string,
   networkIds: NetworkId[] = [],
 ): Promise<TokensInfo> {
-  // Get base tokens
-  const url = networkIds.length
-    ? `${getTokensInfoUrl}?networkIds=${networkIds.join(',')}`
-    : getTokensInfoUrl
-  const data = await got.get(url).json<Record<string, RawTokenInfo>>()
-
-  // Map to TokenInfo
-  const tokensInfo: TokensInfo = {}
-  for (const [tokenId, tokenInfo] of Object.entries(data)) {
-    tokensInfo[tokenId] = {
-      ...tokenInfo,
-      priceUsd: toSerializedDecimalNumber(tokenInfo.priceUsd ?? 0),
-      // We don't have this info here but it's not yet needed for base tokens anyway
-      balance: toDecimalNumber(0n, 0),
-      totalSupply: toDecimalNumber(0n, 0),
+  let tokensInfo = tokensInfoCache.get(TOKENS_INFO_CACHE_KEY) as TokensInfo
```

Comment:
The cache key should include network ids. Otherwise w/ different apps calling with different network ids, we'd return the wrong list of tokens.

Also, if we just want caching for the usdc dapp, we could pass an optional query parameter and only cache if its set. Would also solve the above problem as network ids from the dapp won't change